exec/lifecycle/src/runLifecycleHook.ts

```diff
@@ -94,13 +97,15 @@ Please unset the script-shell option, or configure it to a .exe instead.
   const logLevel = (opts.stdio !== 'inherit' || opts.silent)
     ? 'silent'
     : undefined
+  const modifyBinPaths: ModifyBinPaths = opts.modifyBinPaths ?? ((_, extraBinPaths) => extraBinPaths)
+  const extraBinPaths: string[] = await modifyBinPaths(manifest, opts.extraBinPaths ?? [])
```

Comment:
what if multiple projects will install the same node.js version at the same time? 

I think we should change extraBinPaths in runLifecycleHooksConcurrently before any scripts run.

Reply:
https://github.com/pnpm/pnpm/pull/8277/commits/3c6bdb2fbf026cee3e63c02af1aa5fbf76a85557